pkg/trace/agent/agent.go

```diff
@@ -179,6 +180,10 @@ func (a *Agent) Process(t pb.Trace) {
 		return
 	}
 
+	defer func(start time.Time) {
+		metrics.Timing("datadog.trace_agent.debug.process_trace_ms", time.Since(start), nil, 1)
```

Comment:
Nit: do we need the word `debug` in any of these metric names? Pretty much  all the metrics we have are for "debugging" in some way so may be simpler to just drop that and call it `datadog.trace_agent.process_trace_ms`?

Reply:
> do we need the word debug in any of these metric names?

Whether it's `debug` or something else, it should be clear that this subset of metrics is a namespace which is not expected to be stable and is not for users, but purely for (Datadog) developers. These metrics are subject to being changed or deleted based on potential future code refactors. That's why I didn't want to group them with the others.

Generally, metrics in the trace-agent shouldn't be changed or deleted over night. Customers and support will feel inconveniences if that happens. Thus, we have to separately namespace these. If you have a better name, I'm open to it.

> Pretty much all the metrics we have are for "debugging" in some way so may be simpler to just drop that and call it 

That's true. But I wouldn't just drop it.

Reply:
Meh, there's lots of other metrics that don't have `debug` in them that are really only relevant for someone working on the agent itself (estimated vs compressed bytes, splits, heap, ?), so it's not clear why these specific metrics should be deemed "debug" while those others aren't. Not a big deal, just not clear to me.